standalone-metastore/metastore-common/src/main/java/org/apache/hadoop/hive/metastore/IMetaStoreClient.java

```diff
@@ -3008,6 +3007,15 @@ Function getFunction(String catName, String dbName, String funcName)
   List<String> getFunctions(String dbName, String pattern)
       throws MetaException, TException;
 
+  /**
+   * Get all functions matching a pattern
+   * @param dbName database name.
+   * @param pattern to match.  This is a java regex pattern.
+   * @throws MetaException error accessing the RDBMS
+   * @throws TException thrift transport error
+   */
+  List<Function> getFunctionsInDb(String dbName, String pattern)
```

Comment:
how about change it to `GetFunctionsResponse getFunctions(..` , and mark the `List<String> getFunctions` as deprecated?

Reply:
Ack